10 domain model handling

kenmeacham opened the pull request:
See #10 for further details.

kenmeacham commented:
@scp93ch - seems that I need to have a look at the unit tests, which must have broken with a recent fix..

If you're ready to look at these changes, you could make a start in parallel, while I'm looking into this.

kenmeacham commented:
@scp93ch - unit tests are now fixed so pipelines/checks now pass :)

Ready for review when you are available..

scp93ch commented on src/main/java/uk/ac/soton/itinnovation/security/systemmodeller/rest/ModelController.java:
```diff
@@ -511,30 +508,29 @@ public ResponseEntity<Map<String, RiskLevelCount>> getModelRiskVector(@PathVaria
+	@RequestMapping(value = "/images/{domainModel}/{image}" , method = RequestMethod.GET) 
+    public ResponseEntity<FileSystemResource> getImage(@PathVariable String domainModel, @PathVariable String image) throws IOException {
+		String imageFile = kbInstallFolder + File.separator + domainModel + File.separator + "icons" + File.separator + image;
+		Path path = new File(imageFile).toPath();
```
Okay, but would be better to use java.nio.file.Paths helper.

scp93ch reviewed (changes requested):
Thanks Ken, great job.

PaletteGenerator has unused `storeModelManager` and a large `main` method that is now all commented out and can surely be deleted?

Why do we still need the `resources/static/data` folder?

`DomainModelControllerTest.java` ignores 3 tests saying that "PaletteGenerator assumes that it should read ontologies.json", but the code for `ontologies.json` is commented out in `PaletteGenerator.java`. Can we re-enable the tests?

There are unused imports in `CommonTestSetup.java` (not introduced in this PR).

We don't need `NEXUS_USER/PASS` in `docker-compose.yml` now.

I expected a persistent volume to be added to `docker-compose.yml` for the installed knowledgebases. Are they somehow persisted?

kenmeacham commented:
@scp93ch - unit tests are now fixed so pipelines/checks now pass :)

Ready for review when you are available..

kenmeacham commented:
I fixed some additional issues that I found (see also commits above)

1) Ensure that installed kb (domain model) folders are removed when reset.on.start=true, prior to (re)installing domain zipfiles
   (previously it would fail, claiming that the domain model alredy exists; the reset.on.start=true should blat everything)
2) Ensure that domain model installation folder is also deleted, when deleting a domain model (e.g. via GUI). Previously only the domain model itself was deleted from the triplestore.

kenmeacham commented:
> Thanks Ken, great job.

Thank you! It has been quite a complex issue..

> PaletteGenerator has unused `storeModelManager` and a large `main` method that is now all commented out and can surely be deleted?

Removed (see commit)

> Why do we still need the `resources/static/data` folder?

You're right - we don't any longer, now that palettes, etc, are stored elsewhere. The folder has been removed by deleting its contained .gitignore file

> `DomainModelControllerTest.java` ignores 3 tests saying that "PaletteGenerator assumes that it should read ontologies.json", but the code for `ontologies.json` is commented out in `PaletteGenerator.java`. Can we re-enable the tests?

I've re-instated and fixed the tests for the current behaviour.

> There are unused imports in `CommonTestSetup.java` (not introduced in this PR).

Removed.

> We don't need `NEXUS_USER/PASS` in `docker-compose.yml` now.

I'm confused about this, as there is no NEXUS_USER/PASS in `docker-compose.yml`. If anything, these were replaced by MAVEN_* recently, but even these have been removed now.

> I expected a persistent volume to be added to `docker-compose.yml` for the installed knowledgebases. Are they somehow persisted?

Yes, there is a persistent volume  in `docker-compose.yml`. See line 24:

```
      # Persistent named volume for storage of knowledgebase data (domain model, images and palettes)
      - type: volume
        source: knowledgebases
        target: /opt/spyderisk/knowledgebases
```

Are you looking at the correct file?

scp93ch reviewed (approved):
Looks good Ken. I must have got confused about the docker-compose file.
